### PR TITLE
[ColumnMapping] Add column mapping metadata update functionality

### PR DESCRIPTION
**Pull request by vkorukanti**
(This is third PR for column mapping support in Standalone. End-2-end prototype is at #453)

* Add column mapping modes (`DeltaColumnMappingMode`)
* Add a method to verify and update column mapping metadata (`DeltaColumnMapping.verifyAndUpdateMetadataChange`)

### CODE REVIEW

File: standalone/src/main/scala/io/delta/standalone/internal/DeltaConfig.scala [hunk 1]
```diff
@@ -250,4 +250,31 @@ private[internal] object DeltaConfigs extends Logging {
...
+   * be set by users.
+   */
+  val COLUMN_MAPPING_MAX_ID = buildConfig[Long](
+    "columnMapping.maxColumnId",
+    "0",
+    _.toLong,
+    _ => true,
+    "",
+    minimumProtocolVersion = Some(
+      new Protocol(
```
Comment (scottsand-db) [inline standalone/src/main/scala/io/delta/standalone/internal/DeltaConfig.scala:276]: why not just `DeltaColumnMapping.MIN_PROTOCOL_VERSION`

File: standalone/src/main/scala/io/delta/standalone/internal/actions/actions.scala [hunk 2]
```diff
@@ -241,6 +241,15 @@ private[internal] case class Metadata(
...
     new StructType(partitionColumns.map(c => schema.get(c)).toArray)
 
+  @JsonIgnore
+  lazy val columnMappingMode: DeltaColumnMappingMode =
+    DeltaConfigs.COLUMN_MAPPING_MODE.fromMetadata(this)
+
+  /** Column mapping max id used so far in this table */
+  @JsonIgnore
+  lazy val columnMappingMaxId: Long =
+  DeltaConfigs.COLUMN_MAPPING_MAX_ID.fromMetadata(this)
```
Comment (scottsand-db) [inline standalone/src/main/scala/io/delta/standalone/internal/actions/actions.scala:251]: nit: indentation

**Review by scottsand-db — APPROVED** [review]
LGTM left 2 minor comments

File: standalone/src/main/scala/io/delta/standalone/internal/DeltaConfig.scala (hunk 1 above)
Comment (vkorukanti) [reply standalone/src/main/scala/io/delta/standalone/internal/DeltaConfig.scala:276]: Good idea, changed it

**Pull request merged**